### PR TITLE
Performance of basic vector converts fixed

### LINKED ISSUE #1303: Performance of converts could be speeded up significantly by caching unit type
#### Description:

Currently on every unit convert `BHoMUnitType` method is executed, resulting in redundant 2 `foreach` loops and one Linq query that seem to take up majority of the entire execution time. This becomes an issue in case of performance-sensitive converts. Quick solution is available - it should be relatively simple to cache units for each unit type instead of querying them every time.

Another level of optimisation would be to store the actual conversion factors as doubles, but this would make the code less readable and the change more painful - not sure if it is worth, depends on how much will the performance improve.

### PR DESCRIPTION
**Pull request by pawelbaran**
### Issues addressed by this PR

Partial solution to #1303

### CODE REVIEW

**Comment by pawelbaran** [comment]
@BHoMBot check serialisation
@BHoMBot check null-handling

File: Revit_Core_Engine/Convert/Geometry/FromRevit/Point.cs [hunk 1]
```diff
@@ -53,9 +53,17 @@ public static oM.Geometry.Point PointFromRevit(this XYZ xyz)
...
+            return new oM.Geometry.Point { X = xyz.X * m_LengthToSi, Y = xyz.Y * m_LengthToSi, Z = xyz.Z * m_LengthToSi };
         }
 
+
+        /***************************************************/
+        /****              Private fields               ****/
+        /***************************************************/
+
+        // Optimisation to avoid calling ToSI(SpecTypeId.Length) every time as it causes massive overhead
+        private static double m_LengthToSi = 1.0.ToSI(SpecTypeId.Length);
```
Comment (vietle-bh) [inline Revit_Core_Engine/Convert/Geometry/FromRevit/Point.cs:65]: Would look more consistent...
```suggestion
        private static double m_LengthToSI = 1.0.ToSI(SpecTypeId.Length);
```

File: Revit_Core_Engine/Convert/Geometry/ToRevit/XYZ.cs [hunk 2]
```diff
@@ -47,9 +47,17 @@ public static XYZ ToRevit(this oM.Geometry.Point point)
...
+            return new XYZ(vector.X * m_LengthFromSi, vector.Y * m_LengthFromSi, vector.Z * m_LengthFromSi);
         }
 
+
+        /***************************************************/
+        /****              Private fields               ****/
+        /***************************************************/
+
+        // Optimisation to avoid calling FromSI(SpecTypeId.Length) every time as it causes massive overhead
+        private static double m_LengthFromSi = 1.0.FromSI(SpecTypeId.Length);
```
Comment (vietle-bh) [inline Revit_Core_Engine/Convert/Geometry/ToRevit/XYZ.cs:59]: Would look more consistent...
```suggestion
        private static double m_LengthFromSI = 1.0.FromSI(SpecTypeId.Length);
```

**Review by vietle-bh — COMMENTED** [review]
Huge performance improvement ❤️ 

image

File: Revit_Core_Engine/Convert/Geometry/FromRevit/Point.cs (hunk 1 above)
Comment (pawelbaran) [reply Revit_Core_Engine/Convert/Geometry/FromRevit/Point.cs]: Not sure if using multiple capitals in a row is the right convention, but we did it so far, so applied the change to keep convention.

**Comment by pawelbaran** [comment]
@BHoMBot check core
@BHoMBot check compliance

**Comment by pawelbaran** [comment]
@BHoMBot check core
@BHoMBot check compliance

**Comment by pawelbaran** [comment]
@BHoMBot check serialisation
@BHoMBot check null-handling
@BHoMBot check versioning
@BHoMBot check installer

**Review by vietle-bh — APPROVED** [review]
Works well with great naming 👍

**Comment by FraserGreenroyd** [comment]
@BHoMBot check versioning

**Comment by FraserGreenroyd** [comment]
@BHoMBot check ready-to-merge

**Comment by FraserGreenroyd** [comment]
@BHoMBot check ready-to-merge

**Pull request merged**